gtwrap/matlab_wrapper/wrapper.py

```diff
@@ -137,6 +139,35 @@ def _insert_spaces(self, x, y):
         """
         return x + '\n' + ('' if y == '' else '  ') + y
 
+    @staticmethod
+    def _expand_default_arguments(method, save_backup=True):
+        """Recursively expand all possibilities for optional default arguments.
+        We create "overload" functions with fewer arguments, but since we have to "remember" what
+        the default arguments are for later, we make a backup.
+        """
+        def args_copy(args):
+            return ArgumentList([copy.copy(arg) for arg in args.list()])
+        def method_copy(method):
+            method2 = copy.copy(method)
+            method2.args = args_copy(method.args)
+            method2.args.backup = method.args.backup
+            return method2
+        if save_backup:
+            method.args.backup = args_copy(method.args)
+        method = method_copy(method)
+        for arg in method.args.list():
+            if arg.default is not None:
+                arg.default = None
+                methodWithArg = method_copy(method)
+                method.args.list().remove(arg)
+                return [
+                    *MatlabWrapper._expand_default_arguments(methodWithArg, save_backup=False),
```

Comment:
I am pretty sure you can call `this` instead of `MatlabWrapper`. Let me double check though.

Reply:
it's a static function so it doesn't have a `self` variable, unless there's a keyword `this` that I didn't know about in python?

Reply:
Oops didn't see that. Fair enough, but then what is the reason for this being a staticmethod in the first place?

Reply:
it doesn't need member variables?

I guess when I write python code I tend to make everything static unless it needs member functions/variables, but I guess it's a style thing?  Just felt more functional and "proper" to me.

I can change it if you want but also I don't really see a problem with it

Reply:
You're still calling a class method. If the class method needs an instance variable then the calling method shouldn't be static. If this method and it's constituents don't depend on any instance variable, then you could make this a classmethod instead. Same deal, but you can then use cls to access the method internally which makes semantic sense.

Reply:
I don't follow what you mean...

1. no I'm calling a _static_ method, which is different than a _class_ method in python
2. the method does _not_ need an instance variable, if it did then this wouldn't work?
3. sure I could make this a class method but why?  I think this is a pretty standard usage of staticmethod and this function doesn't need access to any class state or anything (e.g. class static variables) so I think staticmethod is more appropriate here?  basically this function could be outside of the class entirely and it would behave exactly the same, I just included it inside the class because it semantically makes sense to be there.  I feel like that's pretty much the definition of a static function

I kind of feel like this is a pretty minor detail to get hung up on?

Reply:
It is not major for sure, but I wanted to double check. There's some misunderstanding here: I wanted to clarify that expand_default_arguments wasn't using an instance variable in some way.

Generally static methods are used when you just want encapsulation of some functionality. Since you are calling a class method inside this method, it seems more appropriate to do it as a classmethod so you can do cls.expand_default_arguments.

But yeah this is minor and it doesn't make any difference whether it is updated or not.

Reply:
Just to clarify, which class method is getting called from inside this method?  _expand_default_arguments is a static method so it's still calling a static method?

feel free to resolve if you feel ok with it